### ISSUE TITLE
branch has 0 operands for successor #0, but target block has 1 for eir.invoke_global_dynamic

### ISSUE THREAD

**Issue by KronicDeth** [lumen/lumen#668]
While my `eir.call_global_dynamic` from #667 seems to work fine for all tests, 5 instances of `eir.invoke_global_dynamic` in the `lumen/otp` tests are invalid because the branch target block operands disagree

## `lib/megaco/src/binary/megaco_ber_encoder.erl`

<details>

```
020-10-08T02:03:40.1781880Z ---- lumen::otp::lib::megaco::binary::megaco_ber_encoder stdout ----
2020-10-08T02:03:40.1783240Z thread 'lumen::otp::lib::megaco::binary::megaco_ber_encoder' panicked at 'Compilation failed
2020-10-08T02:03:40.1783730Z Commands:
2020-10-08T02:03:40.1784110Z cd /Users/runner/work/lumen/lumen/native_implemented/otp
2020-10-08T02:03:40.1786200Z "../../bin/lumen" "compile" "--output-dir" "tests/external/lumen/otp/lib/megaco/binary/megaco_ber_encoder/_build" "--output" "tests/external/lumen/otp/lib/megaco/binary/megaco_ber_encoder/bin/megaco_ber_encoder" "-O0" "--emit=all" "/Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_ber_encoder.erl" "/Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl" "--include" "/Users/runner/work/lumen/otp/lib"
2020-10-08T02:03:40.1787690Z 
2020-10-08T02:03:40.1787910Z stdout: 
2020-10-08T02:03:40.1788370Z stderr:    Compiling /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_ber_encoder.erl
2020-10-08T02:03:40.1789060Z     Compiled /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_ber_encoder.erl
2020-10-08T02:03:40.1789750Z    Compiling /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl
2020-10-08T02:03:40.1790370Z error: branch has 0 operands for successor #0, but target block has 1
2020-10-08T02:03:40.1791390Z    ┌─ /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl:89:1
2020-10-08T02:03:40.1792100Z    │    
2020-10-08T02:03:40.1792830Z 89 │ ╭ ╭     case (catch TransMod:tr_message(MegaMsg, encode, EC)) of
2020-10-08T02:03:40.1793580Z 90 │ │ │     {'EXIT', Reason} ->
2020-10-08T02:03:40.1794320Z    │ ╰─│^ problem during codegen related to this source code
2020-10-08T02:03:40.1795670Z    │   ╰' see current operation: %186 = "eir.invoke_global_dynamic"(%arg3, %185, %arg1, %183, %arg0)[^bb81, ^bb82] {operand_segment_sizes = dense<[1, 1, 3, 0, 0]> : vector<5xi32>} : (!eir.term, !eir.term, !eir.term, !eir.term, !eir.term) -> !eir.term
2020-10-08T02:03:40.1796400Z     
2020-10-08T02:03:40.1796550Z 
2020-10-08T02:03:40.1796940Z error: module verification failed for megaco_binary_encoder_lib
2020-10-08T02:03:40.1797280Z 
2020-10-08T02:03:40.1797700Z       Failed /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl
2020-10-08T02:03:40.1798080Z 
2020-10-08T02:03:40.1798300Z Status code: 101
2020-10-08T02:03:40.1799100Z Signal: ', native_implemented/otp/tests/test.rs:131:5
2020-10-08T02:03:40.1799410Z 
```

</details>

## `lib/megaco/src/binary/megaco_binary_encoder_lib.erl`
   
<details>

```
2020-10-08T02:03:40.1800400Z ---- lumen::otp::lib::megaco::binary::megaco_binary_encoder_lib stdout ----
2020-10-08T02:03:40.1801660Z thread 'lumen::otp::lib::megaco::binary::megaco_binary_encoder_lib' panicked at 'Compilation failed
2020-10-08T02:03:40.1802170Z Commands:
2020-10-08T02:03:40.1802550Z cd /Users/runner/work/lumen/lumen/native_implemented/otp
2020-10-08T02:03:40.1804460Z "../../bin/lumen" "compile" "--output-dir" "tests/external/lumen/otp/lib/megaco/binary/megaco_binary_encoder_lib/_build" "--output" "tests/external/lumen/otp/lib/megaco/binary/megaco_binary_encoder_lib/bin/megaco_binary_encoder_lib" "-O0" "--emit=all" "/Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl" "--include" "/Users/runner/work/lumen/otp/lib"
2020-10-08T02:03:40.1805570Z 
2020-10-08T02:03:40.1805780Z stdout: 
2020-10-08T02:03:40.1806270Z stderr:    Compiling /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl
2020-10-08T02:03:40.1806960Z error: branch has 0 operands for successor #0, but target block has 1
2020-10-08T02:03:40.1807950Z    ┌─ /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl:70:1
2020-10-08T02:03:40.1808670Z    │    
2020-10-08T02:03:40.1809370Z 70 │ ╭ ╭     case (catch AsnMod:decode('MegacoMessage', Binary)) of
2020-10-08T02:03:40.1810050Z 71 │ │ │     {ok, M} ->
2020-10-08T02:03:40.1810740Z    │ ╰─│^ problem during codegen related to this source code
2020-10-08T02:03:40.1812190Z    │   ╰' see current operation: %87 = "eir.invoke_global_dynamic"(%102, %86, %84, %arg1)[^bb30, ^bb32] {operand_segment_sizes = dense<[1, 1, 2, 0, 0]> : vector<5xi32>} : (!eir.term, !eir.term, !eir.term, !eir.term) -> !eir.term
2020-10-08T02:03:40.1812900Z     
2020-10-08T02:03:40.1813050Z 
2020-10-08T02:03:40.1813430Z error: module verification failed for megaco_binary_encoder_lib
2020-10-08T02:03:40.1813800Z 
2020-10-08T02:03:40.1814230Z       Failed /Users/runner/work/lumen/otp/lib/megaco/src/binary/megaco_binary_encoder_lib.erl
2020-10-08T02:03:40.1814800Z 
2020-10-08T02:03:40.1815030Z Status code: 101
2020-10-08T02:03:40.1815780Z Signal: ', native_implemented/otp/tests/test.rs:131:5
2020-10-08T02:03:40.1816070Z 
```
  </details>

## `lib/common_test/src/ct_gen_conn.erl`

<details>

```
2020-10-08T02:03:39.6360910Z ---- lumen::otp::lib::common_test::ct_gen_conn stdout ----
2020-10-08T02:03:39.6362120Z thread 'lumen::otp::lib::common_test::ct_gen_conn' panicked at 'Compilation failed
2020-10-08T02:03:39.6362640Z Commands:
2020-10-08T02:03:39.6363060Z cd /Users/runner/work/lumen/lumen/native_implemented/otp
2020-10-08T02:03:39.6365640Z "../../bin/lumen" "compile" "--output-dir" "tests/external/lumen/otp/lib/common_test/ct_gen_conn/_build" "--output" "tests/external/lumen/otp/lib/common_test/ct_gen_conn/bin/ct_gen_conn" "-O0" "--emit=all" "/Users/runner/work/lumen/otp/lib/common_test/src/ct_gen_conn.erl" "--include" "/Users/runner/work/lumen/otp/lib" "--include" "/Users/runner/work/lumen/otp/lib/common_test/include" "--include" "/Users/runner/work/lumen/otp/lib/common_test/src" "--include" "/Users/runner/work/lumen/otp/lib/kernel/include" "--include" "/Users/runner/work/lumen/otp/lib/snmp/include"
2020-10-08T02:03:39.6367250Z 
2020-10-08T02:03:39.6367500Z stdout: 
2020-10-08T02:03:39.6367970Z stderr:    Compiling /Users/runner/work/lumen/otp/lib/common_test/src/ct_gen_conn.erl
2020-10-08T02:03:39.6368910Z error: branch has 0 operands for successor #0, but target block has 1
2020-10-08T02:03:39.6369970Z     ┌─ /Users/runner/work/lumen/otp/lib/common_test/src/ct_gen_conn.erl:299:1
2020-10-08T02:03:39.6370690Z     │    
2020-10-08T02:03:39.6371330Z 299 │ ╭ ╭ init_gen(Parent,Opts) ->
2020-10-08T02:03:39.6372150Z 300 │ │ │     process_flag(trap_exit,true),
2020-10-08T02:03:39.6373010Z     │ ╰─│^ problem during codegen related to this source code
2020-10-08T02:03:39.6374610Z     │   ╰' see current operation: %81 = "eir.invoke_global_dynamic"(%17, %80, %34, %51, %68)[^bb69, ^bb36] {operand_segment_sizes = dense<[1, 1, 3, 0, 0]> : vector<5xi32>} : (!eir.term, !eir.term, !eir.term, !eir.term, !eir.term) -> !eir.term
2020-10-08T02:03:39.6375430Z     
2020-10-08T02:03:39.6375630Z 
2020-10-08T02:03:39.6375990Z error: module verification failed for ct_gen_conn
2020-10-08T02:03:39.6376280Z 
2020-10-08T02:03:39.6376690Z       Failed /Users/runner/work/lumen/otp/lib/common_test/src/ct_gen_conn.erl
2020-10-08T02:03:39.6377060Z 
2020-10-08T02:03:39.6377280Z Status code: 101
2020-10-08T02:03:39.6378170Z Signal: ', native_implemented/otp/tests/test.rs:131:5
2020-10-08T02:03:39.6378530Z 
```

</details>

## `lib/kernel/src/application_starter.erl`

<details>

```
2020-10-08T02:03:40.0506450Z ---- lumen::otp::lib::kernel::application_starter stdout ----
2020-10-08T02:03:40.0507690Z thread 'lumen::otp::lib::kernel::application_starter' panicked at 'Compilation failed
2020-10-08T02:03:40.0508180Z Commands:
2020-10-08T02:03:40.0508570Z cd /Users/runner/work/lumen/lumen/native_implemented/otp
2020-10-08T02:03:40.0510760Z "../../bin/lumen" "compile" "--output-dir" "tests/external/lumen/otp/lib/kernel/application_starter/_build" "--output" "tests/external/lumen/otp/lib/kernel/application_starter/bin/application_starter" "-O0" "--emit=all" "/Users/runner/work/lumen/otp/lib/kernel/src/application_starter.erl" "--include" "/Users/runner/work/lumen/otp/lib" "--include" "/Users/runner/work/lumen/otp/lib/kernel/include/" "--include" "/Users/runner/work/lumen/otp/lib/kernel/src"
2020-10-08T02:03:40.0512150Z 
2020-10-08T02:03:40.0512360Z stdout: 
2020-10-08T02:03:40.0512840Z stderr:    Compiling /Users/runner/work/lumen/otp/lib/kernel/src/application_starter.erl
2020-10-08T02:03:40.0513480Z error: branch has 0 operands for successor #0, but target block has 1
2020-10-08T02:03:40.0514880Z     ┌─ /Users/runner/work/lumen/otp/lib/kernel/src/application_starter.erl:99:1
2020-10-08T02:03:40.0515690Z     │    
2020-10-08T02:03:40.0516390Z  99 │ ╭ ╭         case catch Mod:start_phase(Phase, Type, PhaseArgs) of
2020-10-08T02:03:40.0517050Z 100 │ │ │         ok ->
2020-10-08T02:03:40.0517740Z     │ ╰─│^ problem during codegen related to this source code
2020-10-08T02:03:40.0519210Z     │   ╰' see current operation: %101 = "eir.invoke_global_dynamic"(%arg3, %100, %arg0, %arg1, %13)[^bb27, ^bb29] {operand_segment_sizes = dense<[1, 1, 3, 0, 0]> : vector<5xi32>} : (!eir.term, !eir.term, !eir.term, !eir.term, !eir.term) -> !eir.term
2020-10-08T02:03:40.0520160Z     
2020-10-08T02:03:40.0520300Z 
2020-10-08T02:03:40.0520680Z error: module verification failed for application_starter
2020-10-08T02:03:40.0521020Z 
2020-10-08T02:03:40.0521430Z       Failed /Users/runner/work/lumen/otp/lib/kernel/src/application_starter.erl
2020-10-08T02:03:40.0521800Z 
2020-10-08T02:03:40.0522020Z Status code: 101
2020-10-08T02:03:40.0522780Z Signal: ', native_implemented/otp/tests/test.rs:131:5
2020-10-08T02:03:40.0523100Z 
```

</details>

## `lib/megaco/src/text/megaco_compact_text_encoder.erl`

<details>

```
2020-10-08T02:03:40.2370340Z ---- lumen::otp::lib::megaco::text::megaco_compact_text_encoder stdout ----
2020-10-08T02:03:40.2371710Z thread 'lumen::otp::lib::megaco::text::megaco_compact_text_encoder' panicked at 'Compilation failed
2020-10-08T02:03:40.2372230Z Commands:
2020-10-08T02:03:40.2372620Z cd /Users/runner/work/lumen/lumen/native_implemented/otp
2020-10-08T02:03:40.2374750Z "../../bin/lumen" "compile" "--output-dir" "tests/external/lumen/otp/lib/megaco/text/megaco_compact_text_encoder/_build" "--output" "tests/external/lumen/otp/lib/megaco/text/megaco_compact_text_encoder/bin/megaco_compact_text_encoder" "-O0" "--emit=all" "/Users/runner/work/lumen/otp/lib/megaco/src/text/megaco_compact_text_encoder.erl" "--include" "/Users/runner/work/lumen/otp/lib" "--include" "/Users/runner/work/lumen/otp/lib/megaco/src/text"
2020-10-08T02:03:40.2376210Z 
2020-10-08T02:03:40.2376420Z stdout: 
2020-10-08T02:03:40.2376920Z stderr:    Compiling /Users/runner/work/lumen/otp/lib/megaco/src/text/megaco_compact_text_encoder.erl
2020-10-08T02:03:40.2377570Z error: branch has 0 operands for successor #0, but target block has 1
2020-10-08T02:03:40.2378600Z     ┌─ /Users/runner/work/lumen/otp/lib/megaco/src/text/megaco_compact_text_encoder.erl:210:1
2020-10-08T02:03:40.2379330Z     │    
2020-10-08T02:03:40.2379980Z 210 │ ╭ ╭     case (catch ParseMod:parse(Tokens)) of
2020-10-08T02:03:40.2380690Z 211 │ │ │     {ok, MegacoMessage} ->
2020-10-08T02:03:40.2381450Z     │ ╰─│^ problem during codegen related to this source code
2020-10-08T02:03:40.2382740Z     │   ╰' see current operation: %3 = "eir.invoke_global_dynamic"(%arg0, %2, %arg1)[^bb12, ^bb16] {operand_segment_sizes = dense<[1, 1, 1, 0, 0]> : vector<5xi32>} : (!eir.term, !eir.term, !eir.term) -> !eir.term
2020-10-08T02:03:40.2383400Z     
2020-10-08T02:03:40.2383550Z 
2020-10-08T02:03:40.2383940Z error: module verification failed for megaco_compact_text_encoder
2020-10-08T02:03:40.2384270Z 
2020-10-08T02:03:40.2384720Z       Failed /Users/runner/work/lumen/otp/lib/megaco/src/text/megaco_compact_text_encoder.erl
2020-10-08T02:03:40.2385120Z 
2020-10-08T02:03:40.2385340Z Status code: 101
2020-10-08T02:03:40.2386060Z Signal: ', native_implemented/otp/tests/test.rs:131:5
2020-10-08T02:03:40.2386360Z 
```

</details>

**Comment by KronicDeth** [comment]
This might be related to #534 as I copied `InvokeGlobalDynamicOp` from `InvokeOp`, so if there is a bug in `InvokeOp` I likely copied it.

**Comment by bitwalker** [comment]
This is likely fixed by #582, just like #534, but I'll wait for you to confirm once we get everything merged

**Comment by bcardarella** [comment]
Per Dec 9 standup, believe to be fixed
